### DISCUSSION TITLE
Does Pydantic have an equivalent to Marshmallow's ValidationError.valid_data

### DISCUSSION THREAD

**Discussion by johtso** [pydantic/pydantic#2797] [Question] [answered]
[Mashmallow allows you to use the data that was valid when a validation error is raised](https://marshmallow.readthedocs.io/en/stable/marshmallow.exceptions.html#marshmallow.exceptions.ValidationError), which is very useful if you want to log any issues, but fail gracefully and use the data that you can.

Is something similar possible with Pydantic?

**Comment by samuelcolvin** [comment]
Yes, but you'll need to call [`validate_model`](https://github.com/samuelcolvin/pydantic/blob/b718e8e62680c2d7a42f1cf946094e1c0fe28c1d/pydantic/main.py#L1010) directly.

**Reply by johtso** [reply]
Hmm.. how does that work? It looks like that function allows you to validate your input data.. but how do you then load it into the model?

You'd still want your model with all its properties/validators that manipulate data/aliases..

Or does validate_model manipulate the data?

**Comment by samuelcolvin** [comment, marked as answer]
Have a look at the code:

https://github.com/samuelcolvin/pydantic/blob/b718e8e62680c2d7a42f1cf946094e1c0fe28c1d/pydantic/main.py#L414-L424

**Reply by johtso** [reply]
That's really helpful, thanks, will have a play with that at some point.
